File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/database/DatabaseConfigurationView.java
package com.mieai.qqbot.admin.database;

import java.time.Instant;

public record DatabaseConfigurationView(
        long revision,
        DatabaseType type,
        String sqlitePath,
        Long busyTimeoutMs,
        String host,
        Integer port,
        String databaseName,
        String username,
        DatabaseSslMode sslMode,
        Long connectTimeoutMs,
        boolean passwordConfigured,
        String databaseProduct,
        String databaseVersion,
        boolean switchInProgress,
        Instant lastSwitchedAt) {
}

File: qqbot-module-database-support/src/main/kotlin/com/mieai/qqbot/admin/database/DatabaseConfigurationView.kt
package com.mieai.qqbot.admin.database

import java.time.Instant

data class DatabaseConfigurationView(
    val revision: Long,
    val type: DatabaseType,
    val sqlitePath: String?,
    val busyTimeoutMs: Long?,
    val host: String?,
    val port: Int?,
    val databaseName: String?,
    val username: String?,
    val sslMode: DatabaseSslMode?,
    val connectTimeoutMs: Long?,
    val passwordConfigured: Boolean,
    val databaseProduct: String?,
    val databaseVersion: String?,
    val switchInProgress: Boolean,
    val lastSwitchedAt: Instant?,
)
